packages/airnode-node/src/workers/cloud-platforms/aws.ts

```diff
@@ -15,12 +17,32 @@ export function spawn(params: WorkerParameters): Promise<WorkerResponse> {
       Payload: JSON.stringify(params.payload),
     };
 
+    const resolve: typeof spawnResolve = (value) => {
+      logger.debug(`Successful Lambda response: ${value}`);
+      spawnResolve(value);
+    };
+
+    const reject: typeof spawnReject = (error) => {
+      logger.debug(`Lambda invocation or execution failed. Response: ${error}`);
+      spawnReject(error);
+    };
+
+    // The Lambda invocation callback populates the error (first argument) only when the invocation fails (e.g. status
+    // code is 400) or the request is not parsed properly by the SDK and can't be invoked. For errors and timeouts that
+    // happen inside the invoked lambda have "data.FunctionError" and "data.Payload.errorMessage" populated instead.
+    // See: https://stackoverflow.com/q/42672023 and https://stackoverflow.com/q/48644093
     lambda.invoke(options, (err, data) => {
-      if (err) {
-        reject(err);
-        return;
-      }
-      resolve(JSON.parse(JSON.parse(data.Payload as string).body) as WorkerResponse);
+      if (err) return reject(err);
+
+      if (data.FunctionError) return reject(data.Payload);
+
+      const goPayload = goSync(() => JSON.parse(data.Payload?.toString() ?? ''));
+      if (!goPayload.success) return reject(goPayload.error);
+
+      const goBody = goSync(() => JSON.parse(goPayload.data.body));
+      if (!goBody.success) return reject(goBody.error);
```

Comment:
Does it make sense to provide a default for when the `Payload` is not present that will immediately fail during parsing? Isn't checking for `Payload` presence explicitly better?

Reply:
I could do `JSON.parse(data.Payload!.toString())` but that suggests that Payload is defined, for which there is no reason. That's why I opted for `?` but `JSON.parse` requires a string argument, so I defaulted to `?? ''` empty string since it will fail on the next validation anyway.

Reply:
I understand you're reasoning. I was just suggesting handling the `Payload` absence explicitly.
```suggestion
      if (!data.Payload) return reject(new Error('Missing payload');

      const goPayload = goSync(() => JSON.parse(data.Payload!.toString()));
      if (!goPayload.success) return reject(goPayload.error);

      const goBody = goSync(() => JSON.parse(goPayload.data.body));
      if (!goBody.success) return reject(goBody.error);
```

Reply:
Sure, I can do that.